src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -308,309 +263,291 @@ namespace FanCoilUnits {
                                                                      cAlphaFields,
                                                                      cNumericFields);
 
-            FanCoilNum = FanCoilIndex;
-
-            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilNum).FieldNames.allocate(NumNumbers);
-            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilNum).FieldNames = "";
-            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilNum).FieldNames = cNumericFields;
+            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilIndex).FieldNames.allocate(NumNumbers);
+            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilIndex).FieldNames = "";
+            state.dataFanCoilUnits->FanCoilNumericFields(FanCoilIndex).FieldNames = cNumericFields;
```

Comment:
I have a suspicion that we could eliminate this extra array of structs of arrays of field names pretty easily.  But OK for right now while addressing cppcheck things.